### PR TITLE
Manage images

### PR DESCRIPTION
**Pull request by devtuananh**
https://dev.framgia.com/issues/184330

### CODE REVIEW

**Comment by devtuananh** [comment]
ready

File: app/views/images/_image.html.erb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<tr>
+  <td><%= image.id %></td>
+  <td><%= image.room_id %></td>
+  <td><%= image_tag(image.image_link, :size => "260x180") %></td>
+  <td><%= l image.created_at, format: :own %></td>
+  <% if current_user&.admin? %>
```
Comment (huongnguyenmta) [inline app/views/images/_image.html.erb]: check trong controller nữa nhé em

**Comment by huongnguyenmta** [comment]
commented

File: app/views/images/_image.html.erb (hunk 1 above)
Comment (devtuananh) [reply app/views/images/_image.html.erb]: Vâng chị

**Comment by devtuananh** [comment]
ready

File: config/routes.rb [hunk 2]
```diff
@@ -9,4 +9,5 @@
   resources :users, except: :destroy
   resources :account_activations, only: :edit
   resources :categories
+  resources :images, only: %i(index new)
```
[the remark this answers is not in the document]
Comment (devtuananh) [reply config/routes.rb]: It not working. :)) 
I edited it
Thank you

File: app/controllers/images_controller.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    else
+      flash[:danger] = t ".not_found"
+      redirect_to images_path
+    end
+  end
+
+  private
+
+  def load_image
+    @image = Image.find_by id: params[:id] || not_found
```
[the remark this answers is not in the document]
Comment (devtuananh) [reply app/controllers/images_controller.rb]: I edited it

**Comment by devtuananh** [comment]
ready

File: config/routes.rb [hunk 4]
```diff
@@ -9,4 +9,5 @@
   resources :users, except: :destroy
   resources :account_activations, only: :edit
   resources :categories
+  resources :images, except: %i(edit update)
```
Comment (mdmahmudurrahman) [inline config/routes.rb]: In the image controller I see only three actions: index, new, destroy. Can find the mistake ? :)

File: app/controllers/images_controller.rb [hunk 5]
```diff
@@ -0,0 +1,32 @@
+class ImagesController < ApplicationController
+  before_action :load_image, except: %i(index create new)
+  before_action :logged_in_user, except: %i(new create)
+  before_action :verify_staff
+
+  def index
+    @images = Image.by_latest.page(params[:page]).per Settings.show
+  end
+
+  def new
```
Comment (mdmahmudurrahman) [inline app/controllers/images_controller.rb]: Can I know what is the purpose of this action (?)

File: app/controllers/images_controller.rb [hunk 6]
```diff
@@ -0,0 +1,32 @@
+class ImagesController < ApplicationController
+  before_action :load_image, except: %i(index create new)
+  before_action :logged_in_user, except: %i(new create)
```
Comment (mdmahmudurrahman) [inline app/controllers/images_controller.rb]: I do not see any create action here!

File: app/controllers/images_controller.rb [hunk 7]
```diff
@@ -0,0 +1,32 @@
+class ImagesController < ApplicationController
+  before_action :load_image, except: %i(index create new)
```
Comment (mdmahmudurrahman) [inline app/controllers/images_controller.rb]: Is there any create action in this controller you added ?

**Comment by mdmahmudurrahman** [comment]
commented

File: app/controllers/images_controller.rb (hunk 7 above)
Comment (devtuananh) [reply app/controllers/images_controller.rb]: Yes, I write mistake it

File: app/controllers/images_controller.rb (hunk 5 above)
Comment (devtuananh) [reply app/controllers/images_controller.rb]: It create image when I create room, then create image of rooms

**Comment by devtuananh** [comment]
ready

File: app/controllers/images_controller.rb (hunk 5 above)
Comment (mdmahmudurrahman) [reply app/controllers/images_controller.rb]: Not clear what you said. Have you implemented creating room functionality yet ?

**Comment by mdmahmudurrahman** [comment]
commented
Please reply the question! .

File: app/controllers/images_controller.rb (hunk 5 above)
Comment (devtuananh) [reply app/controllers/images_controller.rb]: new method will be create one object image.
Now, I can be remove it ?
Mr. Phuc is implementing this function.

File: config/routes.rb (hunk 4 above)
Comment (devtuananh) [reply config/routes.rb]: I edited it

**Comment by devtuananh** [comment]
ready

**Pull request merged**